### ISSUE TITLE
`ocrConfidence` doesn't effect threshold used for ocr matching

### ISSUE THREAD

**Issue by joel-duffie** [nut-tree/nut.js#551]
**Version**
@nut-tree/nut-js@3.1.2
@nut-tree/nl-matcher@2.2.0
@nut-tree/plugin-ocr@1.0.2

**Short overview**
The `screen.config.ocrConfidence` parameter doesn't effect the actual matching threshold.

**Issue occurs on**

- [ ] ~Virtual machine~ (not tested)
- [ ] ~Docker container~ (not tested)
- [x] Dev/Host system

**node version**:
20.8.0

**OS type and version**:
Mac OS Ventura 13.5.2 (22G91)

**Detailed error description**

I have a script, essentially copied from [this sample in the OCR docs](https://nutjs.dev/plugins/ocr#usage-on-screen-text-search).
I replaced the placeholder values with applicable values, but OCR wouldn't work. I was receiving an error:
```
Error: Searching for word-query-theme failed. Reason: 'Error: OCR run failed.
Reason: Error: Failed to detect word theme with config {"lang":["eng"],"partialMatch":false,"caseSensitive":false}. 
Best match is Theme with confidence 0.9623564910888672'
```
If there was a 0.962 confidence-level match on screen, and the required level is 0.8, it should've matched...
So, I thought to set the `screen.config.confidence` level to something like 0.95. Boom! Issue fixed.

I concluded that the `ocrConfidence` parameter isn't actually effecting the OCR matching threshold.

**Full code sample to reproduce**
```js
const { screen, singleWord, sleep, imageResource } = require("@nut-tree/nut-js");
const {configure, Language, LanguageModelType, preloadLanguages} = require("@nut-tree/plugin-ocr");
require('@nut-tree/nl-matcher');
const utils = require('../scripts/rv_utilities');

configure({
    languageModelType: LanguageModelType.BEST,
    dataPath: utils.getLangsPath()
});

(async () => {
    try {
        await preloadLanguages([Language.English]);

        screen.config.ocrConfidence = 0.80;
        screen.config.confidence = 0.85;
        screen.config.autoHighlight = true;
        screen.config.resourceDirectory = utils.getImagesPath('mac');
        
        sleep(3000);
        const reg = await screen.find(imageResource('slideBlueBorderLarge1.png'))

        const word = await screen.findAll(singleWord("Theme"), {
            searchRegion: reg,
            providerData: {
                lang: [Language.English],
                partialMatch: false,
                caseSensitive: false
            }
        });
        word.forEach((w) => {
            console.log('match @', w.left, ',', w.top);
        });

    } catch(err) {
        console.error(err);
    }
})();
```

**Additional content**
![slideBlueBorderLarge1](https://github.com/nut-tree/nut.js/assets/141655392/cfd118e3-b975-4d44-aa33-45503bfd35aa)
this is the image I used to match

> Please provide any (mandatory) additional data to reproduce the error (Dockerfiles etc.)

**Comment by s1hofmann** [comment]
Hi @joel-duffie 👋 

thanks for raising this issue.
This is a known bug which will be fixed in the next release, it just takes some time because it requires plugins to be updated accordingly.

Best regards

Simon

**Comment by s1hofmann** [comment]
Closed with [release 4.0.0](https://github.com/nut-tree/nut.js/releases/tag/v4.0.0)
